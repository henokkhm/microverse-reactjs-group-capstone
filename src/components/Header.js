import Navbar from './Navbar';
import logo from '../assets/logo.png';

function Header() {
  return (
    <header>
      <div>
        <img src={logo} alt="Space Travelers' Hub logo" />
        <h1>Space Travelers&#39; Hub</h1>
      </div>
      <Navbar />
    </header>
  );
}
export default Header;
