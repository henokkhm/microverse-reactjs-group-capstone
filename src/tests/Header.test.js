import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import Header from '../components/Header';

it('The Header component renders as expected', () => {
  const { asFragment } = render(
    <BrowserRouter>
      <Header />
    </BrowserRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
