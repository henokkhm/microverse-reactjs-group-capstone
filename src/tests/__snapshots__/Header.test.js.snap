// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Header component renders as expected 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <div
      class="branding"
    >
      <img
        alt="Space Travelers' Hub logo"
        class="logo"
        src="logo.png"
      />
      <h1
        class="pageTitle"
      >
        Space Travelers' Hub
      </h1>
    </div>
    <nav
      class="navigation"
    >
      <ul
        class="navList"
      >
        <li
          class="navListItem"
        >
          <a
            aria-current="page"
            class="navLink active"
            data-testid="navlink-rockets-page"
            href="/"
          >
            Rockets
          </a>
        </li>
        <li
          class="navListItem"
        >
          <a
            class="navLink"
            data-testid="navlink-missions-page"
            href="/missions"
          >
            Missions
          </a>
        </li>
        <li
          class="navListItem"
        >
          <a
            class="navLink"
            data-testid="navlink-my-profile-page"
            href="/my-profile"
          >
            My Profile
          </a>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
